plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergReadVersionedTable.java

```diff
@@ -0,0 +1,186 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.iceberg;
+
+import com.google.common.collect.ImmutableSet;
+import io.trino.Session;
+import io.trino.plugin.base.CatalogName;
+import io.trino.plugin.hive.HdfsConfig;
+import io.trino.plugin.hive.HdfsConfiguration;
+import io.trino.plugin.hive.HdfsConfigurationInitializer;
+import io.trino.plugin.hive.HdfsEnvironment;
+import io.trino.plugin.hive.HiveHdfsConfiguration;
+import io.trino.plugin.hive.NodeVersion;
+import io.trino.plugin.hive.authentication.NoHdfsAuthentication;
+import io.trino.plugin.hive.metastore.HiveMetastore;
+import io.trino.plugin.hive.metastore.MetastoreConfig;
+import io.trino.plugin.hive.metastore.file.FileHiveMetastore;
+import io.trino.plugin.hive.metastore.file.FileHiveMetastoreConfig;
+import io.trino.plugin.iceberg.catalog.file.FileMetastoreTableOperationsProvider;
+import io.trino.spi.connector.SchemaTableName;
+import io.trino.spi.security.Identity;
+import io.trino.spi.security.SelectedRole;
+import io.trino.spi.type.TestingTypeManager;
+import io.trino.testing.AbstractTestQueryFramework;
+import io.trino.testing.DistributedQueryRunner;
+import org.apache.iceberg.Table;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.time.Instant;
+import java.util.Optional;
+
+import static io.trino.spi.security.SelectedRole.Type.ROLE;
+import static io.trino.testing.TestingConnectorSession.SESSION;
+import static io.trino.testing.TestingSession.testSessionBuilder;
+
+public class TestIcebergReadVersionedTable
+        extends AbstractTestQueryFramework
+{
+    private TrinoHiveCatalog catalog;
+    private long v1Unixtime;
+    private long v1SnapshotId;
+    private long v2Unixtime;
+    private long v2SnapshotId;
+
+    @Override
+    protected DistributedQueryRunner createQueryRunner()
+            throws Exception
+    {
+        Session session = testSessionBuilder()
+                .setIdentity(Identity.forUser("iceberg")
+                        .withConnectorRole("iceberg", new SelectedRole(ROLE, Optional.of("admin")))
+                        .build())
+                .setCatalog("iceberg")
+                .build();
+        DistributedQueryRunner queryRunner = DistributedQueryRunner.builder(session).build();
```

Comment:
Any reason not to reuse the code in `IcebergQueryRunner#createIcebergQueryRunner`?

Reply:
because I need the `catalog` object to perform operations like `loadTable` and inspect information in the table.

Reply:
Please do use a helper method like the following

```
    private Optional<Long> getLatestSnapshotId(String tableName)
    {
        MaterializedResult result = computeActual(format("SELECT snapshot_id FROM \"%s$snapshots\" ORDER BY committed_at DESC LIMIT 1", tableName));
        return result.getRowCount() > 0 ? Optional.of((Long) result.getOnlyValue()) : Optional.empty();
    }
```

and replace the `createQueryRunner()` method with:

```
    @Override
    protected DistributedQueryRunner createQueryRunner()
            throws Exception
    {
        return createIcebergQueryRunner();
    }
```

In this fashion you can avoid the low-level details of building from scratch the `IcebergQueryRunner` and working with Iceberg specific API and the maintenance of the test can be made easier.